controllers/usersignup/usersignup_controller_test.go

```diff
@@ -1532,6 +1795,130 @@ func TestUserSignupWithExistingMUROK(t *testing.T) {
 	})
 }
 
+func TestMigrateExistingMURToSpace(t *testing.T) {
+	customTier := newNsTemplateTier("custom", "dev", "stage")
+
+	for testname, testTier := range map[string]*toolchainv1alpha1.NSTemplateTier{
+		"default tier":     baseNSTemplateTier,
+		"non-default tier": customTier,
+	} {
+		t.Run(testname, func(t *testing.T) {
+			member := NewMemberCluster(t, "member1", v1.ConditionTrue)
+			logf.SetLogger(zap.New(zap.UseDevMode(true)))
+			t.Run("MUR exists and still references NSTemplateSet", func(t *testing.T) {
+				userSignup := NewUserSignup()
+				userSignup.Annotations = map[string]string{
+					toolchainv1alpha1.UserSignupUserEmailAnnotationKey: "foo@redhat.com",
+				}
+				userSignup.Labels = map[string]string{
+					toolchainv1alpha1.UserSignupUserEmailHashLabelKey: "fd2addbd8d82f0d2dc088fa122377eaa",
+					"toolchain.dev.openshift.com/approved":            "true",
+				}
+				userSignup.Spec.OriginalSub = "original-sub:foo"
+
+				mur := newMasterUserRecord(userSignup, "member1", testTier, "foo")
+				mur.Labels = map[string]string{toolchainv1alpha1.MasterUserRecordOwnerLabelKey: userSignup.Name}
+				templates := nstemplateSetFromTier(*testTier)
+				ua := toolchainv1alpha1.UserAccountEmbedded{
+					TargetCluster: "member1",
+					SyncIndex:     "123abc", // default value
+					Spec: toolchainv1alpha1.UserAccountSpecEmbedded{
+						UserAccountSpecBase: toolchainv1alpha1.UserAccountSpecBase{
+							NSLimit:       testTier.Name,
+							NSTemplateSet: &templates,
+						},
+					},
+				}
+				// set the user account
+				mur.Spec.UserAccounts = append(mur.Spec.UserAccounts, ua)
```

Comment:
I would just set only one UserAccount
```suggestion
				mur.Spec.UserAccounts = []toolchainv1alpha1.UserAccountEmbedded{ua}
```